drivers/hmis/spec/system/hmis/form_builder_spec.rb

```diff
@@ -0,0 +1,179 @@
+
+        find("button[aria-label='custom_question_1 move up']").trigger('click')
+        expect('Custom question 1').to appear_before('Assessment Date')
+
+        find("button[type='submit']").trigger('click')
+
+        draft.reload
+        expect(draft.definition.dig('item', 0, 'item', 0, 'text')).to eq('Custom question 1')
+        expect(draft.definition.dig('item', 0, 'item', 1, 'text')).to eq('Assessment Date')
+
+        # todo @martha - add more reorders. up/down into/out of groups
+      end
+
+      it 'deletes item' do
+        assert_text 'Group: Test Custom Assessment'
+        find("div[aria-label='item section_1']").click
+        assert_text 'Date: Assessment Date'
+
+        find("button[aria-label='custom_question_1 item actions']").click
+        find("li[aria-label='delete custom_question_1']").click
+        assert_no_text 'Custom question 1'
+
+        find("button[type='submit']").trigger('click')
+
+        draft.reload
+        expect(draft.definition.dig('item', 0, 'item').size).to eq(1)
+      end
+
+      it 'publishes form' do
+        assert_text 'Group: Test Custom Assessment'
+        click_button 'Preview / Publish'
+        assert_text 'PREVIEWING DRAFT System Test 2'
+        click_button 'Publish'
+        assert_text 'Are you sure you want to publish this form?'
+        click_button 'Confirm'
+        assert_text 'SELECTED FORM System Test 2'
+        assert_text 'Status Published'
+        draft.reload
+        expect(draft.status).to eq('published')
+      end
+    end
+
+    context 'with a draft that has advanced features and non-admin user' do
+      before(:each) do
+        remove_permissions(access_control, :can_administrate_config)
+        visit "/admin/forms/advanced_form/#{advanced_draft.id}/edit"
+      end
+
+      it 'does not clobber custom rule or autofill' do
+        find("div[aria-label='item section_1']").click
+        assert_text 'Conditionally autofilled'
+        find("button[aria-label='conditionally_autofilled item actions']").click
+        find("li[aria-label='edit conditionally_autofilled']").click
+        assert_text 'EDIT FORM ITEM Conditionally autofilled'
+        assert_no_text 'Autofill Value 1' # Autofill condition exists, but this user can't see it
+        assert_no_text 'some-particular-project-id' # Custom rule involving this project ID exists, but this user can't see it
+
+        fill_in 'Helper Text', with: 'some helpful information'
+        find("button[type='submit']").trigger('click')
+        assert_text 'Text: Conditionally autofilled'
+
+        advanced_draft.reload
+        item = advanced_draft.definition.dig('item', 0, 'item', 2)
+        autofill = item.dig('autofill_values', 0)
+
+        expect(autofill.dig('value_code')).to eq('filled')
+        expect(autofill.dig('autofill_when', 0, 'question')).to eq('yes_or_no')
+
+        # todo @martha - this is a bug
+        expect(item.dig('custom_rule', 'value')).to eq('some-particular-project-id')
```

Comment:
@gigxz it's a good thing you called this out on the issue description; it turns out the form builder does clobber `custom_rule` and `rule` and, I believe, any other property that's allowed on the form definition json schema but not present on the `FormItem` graphql type. 

I'm a bit torn between possible approaches,
- we could add all possible properties to the form item [schema object](https://github.com/greenriver/hmis-warehouse/blob/11be216286e6b81d4e1367813d8ea53bb3a69b59/drivers/hmis/app/graphql/types/forms/form_item.rb), even if they may not be edited by the form builder. This could fail and silently start to clobber properties if we were to ever add a new property to the schema and forget to add it to the input object, but that seems unlikely?
- Maybe we should make it the job of the [UpdateFormDefinition mutation](https://github.com/greenriver/hmis-warehouse/blob/11be216286e6b81d4e1367813d8ea53bb3a69b59/drivers/hmis/app/graphql/mutations/update_form_definition.rb) to merge the old and new definitions? But to make sure it doesn't re-merge properties that have been intentionally deleted, we would need a list of properties that are in the json schema but not the FormItem. And then we would be back to needing to remember to add any new properties in 2 places, same concern as above. We could generate this list programmatically, I guess, but I'm not sure that seems worthwhile.

Reply:
Edit: Gig created a new ticket https://github.com/open-path/Green-River/issues/6555, we can consider this out of scope for this PR